paleontographica-abteilung-b-palaeobotany-palaeophytology.csl

```diff
@@ -0,0 +1,246 @@
+                <text variable="URL"/>
+              </group>
+            </group>
+          </if>
+        </choose>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher">
+    <group prefix=" ">
+      <text variable="genre" suffix=": " text-case="capitalize-first"/>
+      <choose>
+        <if type="speech paper-conference" match="none">
+          <group delimiter=" ">
+            <text variable="publisher" prefix=" (" suffix=")"/>
+            <text variable="publisher-place"/>
+          </group>
+        </if>
+      </choose>
+    </group>
+    <choose>
+      <if type="speech paper-conference" match="any">
+        <text variable="event" prefix=" " font-style="italic"/>
+        <text variable="event-place" prefix=" "/>
+        <text variable="publisher" prefix=" (" suffix=")"/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="issued">
+    <choose>
+      <if variable="issued">
+        <group>
+          <date variable="issued">
+            <date-part name="year"/>
+          </date>
+        </group>
+      </if>
+      <else>
+        <text term="no date"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="issued-year">
+    <choose>
+      <if variable="issued">
+        <date variable="issued">
+          <date-part name="year"/>
+        </date>
+      </if>
+      <else>
+        <text term="no date" form="short"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="title">
+    <group delimiter=". ">
+      <text variable="title"/>
+      <choose>
+        <if type="book">
+          <text macro="edition"/>
+        </if>
+      </choose>
+    </group>
+  </macro>
+  <macro name="container-title">
+    <choose>
+      <if type="chapter">
+        <group delimiter=". ">
+          <text variable="container-title"/>
+          <text macro="edition"/>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <macro name="edition">
+    <choose>
+      <if type="bill book chapter graphic legal_case legislation motion_picture paper-conference report song" match="any">
+        <choose>
+          <if is-numeric="edition">
+            <number variable="edition" form="long-ordinal" text-case="capitalize-first"/>
+            <text term="edition" form="long" prefix=" " text-case="capitalize-first"/>
+          </if>
+          <else>
+            <text variable="edition"/>
+          </else>
+        </choose>
+      </if>
+    </choose>
+  </macro>
+  <macro name="number-of-pages">
+    <choose>
+      <if type="book report" match="any">
+        <group delimiter=" ">
+          <label plural="always" strip-periods="false" variable="page" form="short"/>
+          <text variable="number-of-pages"/>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <macro name="locators">
+    <choose>
+      <if type="article-journal article-magazine article-newspaper" match="any">
+        <text variable="container-title" font-style="normal"/>
```

Comment:
I'd remove all the `normal` attributes. They're just clutter and we have occasionally seen them cause problems.